bin/cron/admin_stats

```diff
@@ -0,0 +1,113 @@
+  teachers = users.where(user_type: 'teacher')
+  # Active students.
+  seven_day_active_students = students.where(current_sign_in_at: (Time.now - SEVEN_DAYS)..Time.now)
+  thirty_day_active_students = students.where(
+    current_sign_in_at: (Time.now - THIRTY_DAYS)..Time.now
+  )
+  # Retention of students...
+  # ...for measuring the percentage of students with accounts at least thirty-seven days and logged
+  #    in at least once after thirty days.
+  thirty_day_retention_historical_baseline = students.
+    where(created_at: Time.new(1970, 1, 1)..(Time.now - (THIRTY_DAYS + SEVEN_DAYS)))
+  # ...for measuring the percentage of students that started thirty-seven to forty-four days ago
+  #    and logged in at least once after thirty days.
+  thirty_day_retention_current_weekly_baseline = students.where(
+    created_at: (Time.now - (THIRTY_DAYS + 2 * SEVEN_DAYS))..(Time.now - (THIRTY_DAYS + SEVEN_DAYS))
+  )
+  # ...for measuring the percentage of students that started sixty to ninety days ago and logged in
+  #    at least once after thirty days
+  thirty_day_retention_current_monthly_baseline = students.where(
+    created_at: (Time.now - (3 * THIRTY_DAYS))..(Time.now - (2 * THIRTY_DAYS))
+  )
+
+  Properties.set(
+    :admin_stats,
+    {
+      accounts: {
+        users: all_users.count,
+        students: all_students.count,
+        teachers: all_teachers.count,
+      },
+      born_accounts: {
+        users: users.count,
+        students: students.count,
+        female_students: students.where(gender: 'f').count,
+        male_students: students.where(gender: 'm').count,
+        teachers: teachers.count,
+      },
+      actives: {
+        seven_day_students: seven_day_active_students.count,
+        seven_day_female_students: seven_day_active_students.where(gender: 'f').count,
+        seven_day_male_students: seven_day_active_students.where(gender: 'm').count,
+        thirty_day_students: thirty_day_active_students.count,
+        thirty_day_female_students: thirty_day_active_students.where(gender: 'f').count,
+        thirty_day_male_students: thirty_day_active_students.where(gender: 'm').count,
+      },
+      thirty_day_retention_historical: build_demographic_breakdowns(
+        thirty_day_retention_historical_baseline
+      ),
+      thirty_day_retention_current_weekly: build_demographic_breakdowns(
+        thirty_day_retention_current_weekly_baseline
+      ),
+      thirty_day_retention_current_monthly: build_demographic_breakdowns(
+        thirty_day_retention_current_monthly_baseline
+      ),
+      # TODO(asher): Add more student-teacher metrics and pipe them to the admin_stats dashboard.
+      teachers: {
+        with_students: students.join(:followers, student_user_id: :id).select(:followers__user_id).distinct.count,
+        with_active_students: thirty_day_active_students.join(:followers, student_user_id: :id).select(:followers__user_id).distinct.count,
+      },
+    }
+  )
+end
+
+main
```

Comment:
Is this file being re-added intentionally? It looks like it was removed in https://github.com/code-dot-org/code-dot-org/pull/14315